Mathlib/Algebra/Divisibility/Ring.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2023 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import Mathlib.Algebra.Divisibility.Basic
+import Mathlib.Data.Nat.Choose.Sum
+import Mathlib.Algebra.BigOperators.Ring
+import Mathlib.Tactic.Linarith
+
+/-!
+# Lemmas about divisibility in rings
+-/
+
+variable {R : Type*} {x y : R}
+
+lemma dvd_nsmul_of_dvd [Semiring R] {n : ℕ} (h : x ∣ y) : x ∣ n • y :=
+  ⟨n • Classical.choose h, by rw [mul_smul_comm, ← Classical.choose_spec h]⟩
+
+namespace Commute
+
+variable {n m p : ℕ} (hp : n + m ≤ p + 1)
+
+section Semiring
+
+variable [Semiring R] (h_comm : Commute x y)
+
+lemma pow_dvd_add_pow_of_pow_eq_zero_right (hy : y ^ n = 0) :
```

Comment:
Given almost everything in this file is about `y ^ n = 0`, I think it's worth mentioning this in the module docstring as one of the things this file is about.

Reply:
Done.